### ISSUE TITLE
vagrant powershell -c hangs indefinitely at inlinescript block when executing a script with a powershell workflow

### ISSUE THREAD

**Issue by seanamosw** [mitchellh/vagrant#7900]
### Vagrant version

1.8.6
### Host operating system

Windows 7
### Guest operating system

Windows Server 2012 R2
### Vagrantfile

``` ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.configure("2") do |config|
  config.vm.box = "wongaza/win2012r2-bedev"
  config.vm.guest = "windows"
  config.vm.communicator = "winrm"

  config.vm.boot_timeout = 600
  config.vm.graceful_halt_timeout = 600

  config.vm.network "private_network", ip: "192.168.50.4"
  config.vm.network :forwarded_port, guest: 3389, host: 33389, id: "rdp", auto_correct: true
  config.vm.network :forwarded_port, guest: 5985, host: 5985, id: "winrm", auto_correct: true

  config.vm.provision "shell", path: "provision/remotedebug.ps1", :run => "always"

  config.vm.provider "virtualbox" do |vb, override|
     vb.cpus = 4
     vb.memory = 4096
  end
end
```
### Expected behavior

Script should execute to completion as it does when directly executed in the guest.
### Actual behavior

The script hangs indefinitely once an inlinescript block has been reached.
### Steps to reproduce
1. Create a script with a powershell workflow with and inlinescript block eg:
   test.ps1:
   
   ``` powershell
   workflow test {
       inlinescript { Write-Output "In inlinescript block" }
   }
   
   test
   ```
2. Copy or share the script to the guest.
3. Execute the script:
   `powershell
   vagrant powershell -c ".\test.ps1"
   `
### Notes

The same script executes correctly when using a powershell session:

``` powershell
vagrant powershell
.\test.ps1
```

**Comment by chrisroberts** [comment]
Should be fixed by #7875

**Comment by chrisroberts** [comment]
Hi there,

It looks like this has been resolved within a previously shipped version of Vagrant so I am now closing this issue. If the original issue was not fully resolved, please reopen this issue or create a new one.

Cheers!
